web/src/input_pill.ts

```diff
@@ -262,8 +262,21 @@ export function create<ItemType extends {type: string}>(
         },
 
         insertManyPills(pills: string | string[]) {
-            if (typeof pills === "string") {
+            if (typeof pills === "string" && store.split_text_on_comma) {
                 pills = pills.split(/,/g).map((pill) => pill.trim());
+            } else {
+                const parsed_emails = parseAddressList(typeof pills === "string" ? pills : "");
```

Comment:
Can't we just call this if `typeof pills === "string"`, since it does not do anything if it is a list of strings.